src/app/client/src/app/modules/learn/components/course-consumption/course-player/course-player.component.ts

```diff
@@ -228,6 +231,18 @@ export class CoursePlayerComponent implements OnInit, OnDestroy {
     });
   }
 
+  checkExtUrl() {
```

Comment:
The same implementation is repeated in three places. We should avoid it.

Reply:
Done